### PR TITLE
Issue 83: Reorganize repository structure targeting develop branch

### LINKED ISSUE #83: Reorganize repository structure
**Problem description**
Currently, the repository lacks from a clear structure that distinguishes between components (standalone, flink, hadoop), features (state synchronizer) and scenarios (IoT, anomaly detection).

**Problem location**
Repository structure.

**Suggestions for an improvement**
Reorganize the repository structure in a way that can be cleanly extended with potentially diverse types of examples.

### PR DESCRIPTION
**Pull request by RaulGracia**
**Change log description**
This PR ports the changes from #94 to `develop` branch. 

This PR represents a major change in how sample applications are organized. A general overview of the new structure can be seen in this [wiki page](https://github.com/pravega/pravega-samples/wiki/Repository-Structure-Reorganization). In summary, existing samples have been organized as follows:

pravega-client-examples
    - gettingstarted
    - consolerw
    - noop
    - statesynchronizer
    - streamcuts

flink-connector-examples
    - wordcount
    - primer

hadoop-connector-examples
    - wordcount

scenarios
    - turbine-heat-sensor	
    - turbine-heat-processor
    - anomaly-detection

The idea is to have simple and small examples within `xxx-examples` projects based on the type of example at hand, whereas more complex, potentially polyglot examples are organized as separate projects in the `scenarios` folder. This change also implied to modify accordingly the `build` and `readme` files.

**Purpose of the change**
Fixes #83.

**What the code does**
Change the structure of samples repository. Note that I ported the existing code from #94.

**How to verify it**
Applications should run as before with the new repository structure.

### CODE REVIEW

File: hadoop-connector-examples/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ apply plugin: "distribution"
 
 sourceCompatibility = 1.8
 archivesBaseName = 'pravega-hadoop-examples'
-version = connectorVersion
+version = hadoopConnectorVersion
```
Comment (vijikarthi) [inline hadoop-connector-examples/build.gradle]: It should be samples version. we can remove this variable assignment. I have opened a PR to address the hadoop connector version mismatch issue (https://github.com/pravega/pravega-samples/pull/106).

File: scenarios/anomaly-detection/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+### dependencies
+flinkConnectorVersion=0.3.0-102.0e30f47-SNAPSHOT
```
Comment (vijikarthi) [inline scenarios/anomaly-detection/gradle.properties]: Do we need a new file at the module level? Can we not make use of the root gradle properties file?

File: scenarios/turbine-heat-processor/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+## Execution
+Run the sample from the command-line:
+```
+$ bin/run-example [--controller <URI>] [--input <scope>/<stream>] [--startTime <long>] [--output <path>]
+```
+
+Alternately, run the sample from the Flink UI.
+- JAR: `pravega-flink-examples-0.1.0-SNAPSHOT-all.jar`
```
Comment (vijikarthi) [inline scenarios/turbine-heat-processor/README.md]: Can we rename the file name to `pravega-flink-examples-<VERSION>-all.jar`?

File: scenarios/turbine-heat-processor/gradle.properties [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+### dependencies
+flinkConnectorVersion=0.3.0-102.0e30f47-SNAPSHOT
```
Comment (vijikarthi) [inline scenarios/turbine-heat-processor/gradle.properties]: Do we need a new file at the module level? Can we not make use of the root gradle properties file?

File: scenarios/turbine-heat-processor/src/main/dist/bin/run-example [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#!/usr/bin/env bash
+
+flink run -c io.pravega.turbineheatprocessor.TurbineHeatProcessor lib/pravega-flink-examples-0.1.0-SNAPSHOT-all.jar $*
```
Comment (vijikarthi) [inline scenarios/turbine-heat-processor/src/main/dist/bin/run-example]: I believe the version name will be different? `pravega-flink-examples-0.1.0-SNAPSHOT-all.jar`

File: scenarios/turbine-heat-sensor/gradle.properties [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+### dependencies
+junitVersion=4.11
+commonsCLIVersion=1.4
+pravegaVersion=0.3.0-50.f56b52d-SNAPSHOT
```
Comment (vijikarthi) [inline scenarios/turbine-heat-sensor/gradle.properties]: do we need a separate file at the module level? Can we not make use of root gradle.properties configuration?

File: flink-connector-examples/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+## Pre-requisites
+1. Pravega running (see [here](http://pravega.io/docs/latest/getting-started/) for instructions)
+2. Build [pravega-samples](https://github.com/pravega/pravega-samples) repository
+3. Apache Flink running
+
+
+### Distributing Flink Samples
+#### Assemble
+Use gradle to assemble a distribution folder containing the Flink programs as a ready-to-deploy 
+uber-jar called `pravega-flink-examples-0.1.0-SNAPSHOT-all.jar`:
```
[the remark this answers is not in the document]
Comment (RaulGracia) [reply flink-connector-examples/README.md]: Changed.

File: flink-connector-examples/README.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+$ ./gradlew installDist
+...
+$ ls -R flink-connector-examples/build/install/pravega-flink-examples
+bin	lib
+
+flink-connector-examples/build/install/pravega-flink-examples/bin:
+run-example
+
+flink-connector-examples/build/install/pravega-flink-examples/lib:
+pravega-flink-examples-0.1.0-SNAPSHOT-all.jar
```
[the remark this answers is not in the document]
Comment (RaulGracia) [reply flink-connector-examples/README.md]: Changed.

File: flink-connector-examples/README.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+#### Upload
```
[the remark this answers is not in the document]
Comment (RaulGracia) [reply flink-connector-examples/README.md]: I agree if you want to remove this in a separate issue, probably will reduce complexity for the user.

File: hadoop-connector-examples/build.gradle (hunk 1 above)
Comment (RaulGracia) [reply hadoop-connector-examples/build.gradle]: Ok, thanks.

File: scenarios/anomaly-detection/gradle.properties (hunk 2 above)
Comment (RaulGracia) [reply scenarios/anomaly-detection/gradle.properties]: Removed `gradle.properties` from scenarios sub-projects. Now all properties are in the root gradle properties file.

File: scenarios/turbine-heat-processor/README.md (hunk 3 above)
Comment (RaulGracia) [reply scenarios/turbine-heat-processor/README.md]: Changed.

File: scenarios/turbine-heat-processor/gradle.properties (hunk 4 above)
Comment (RaulGracia) [reply scenarios/turbine-heat-processor/gradle.properties]: It has been removed (see previous comment on this issue).

File: scenarios/turbine-heat-processor/src/main/dist/bin/run-example (hunk 5 above)
Comment (RaulGracia) [reply scenarios/turbine-heat-processor/src/main/dist/bin/run-example]: You are right, I changed it.

File: scenarios/turbine-heat-sensor/gradle.properties (hunk 6 above)
Comment (RaulGracia) [reply scenarios/turbine-heat-sensor/gradle.properties]: It has been removed (see previous comment on this issue).

File: flink-connector-examples/README.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+The `upload` task makes it easy to upload the sample binaries to your cluster. First, configure 
+Gradle with the address of a node in your cluster.   Edit `~/.gradle/gradle.properties` to 
+specify a value for `dcosAddress`.
+
+```
+$ cat ~/.gradle/gradle.properties
+dcosAddress=10.240.124.164
```
[the remark this answers is not in the document]
Comment (RaulGracia) [reply flink-connector-examples/README.md]: I agree with this observation. This is related to a previous comment of @vijikarthi that also suggested to remove the entire `Upload` section from this README. He also suggested to do that in a separate PR.

**Pull request merged**